ReSharper.FSharp/src/FSharp.Psi.Features/src/Daemon/Analyzers/IfExpressionAnalyzer.fs

```diff
@@ -0,0 +1,39 @@
+namespace JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.Analyzers
+
+open JetBrains.ReSharper.Feature.Services.Daemon
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.Highlightings.Errors
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Impl
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Parsing
+open JetBrains.ReSharper.Plugins.FSharp.Psi.PsiUtil
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Tree
+open JetBrains.ReSharper.Plugins.FSharp.Util
+
+[<ElementProblemAnalyzer([| typeof<IIfThenElseExpr> |],
+                         HighlightingTypes = [| typeof<IfCanBeReplacedWithConditionOperandWarning> |])>]
+type IfExpressionAnalyzer() =
+    inherit ElementProblemAnalyzer<IIfThenElseExpr>()
+
+    override this.Run(expr, _, consumer) =
+        let thenExpr = expr.ThenExpr
+        let elseExpr = expr.ElseExpr
+
+        match thenExpr.IgnoreInnerParens(), elseExpr.IgnoreInnerParens() with
+        | :? ILiteralExpr as thenLiteral, (:? ILiteralExpr as elseLiteral) ->
+            let thenToken = getTokenType thenLiteral.Literal
```

Comment:
`token` name for a token type is misleading. Especially, since we work with `then` token just a few lines below, and this then-expression-token-type is actually not the then-token.